File: sdkwork-ai-sdk-flutter/lib/src/api/knowledge_bases.dart
import '../http/client.dart';
import '../models.dart';

class KnowledgeBasesApi {
  final HttpClient _client;
  
  KnowledgeBasesApi(this._client);

  /// List documents
  Future<ListDocumentsResponse?> listDocuments(String knowledgeBaseId, Map<String, dynamic>? params) async {
    final response = await _client.get(ApiPaths.aiPath('/knowledge-bases/${knowledgeBaseId}/documents'), params: params);
    return response is ListDocumentsResponse ? response : null;
  }

  /// Add document
  Future<DocumentResponse?> addDocument(String knowledgeBaseId, DocumentAddRequest body) async {
    final response = await _client.post(ApiPaths.aiPath('/knowledge-bases/${knowledgeBaseId}/documents'), body: body, contentType: 'application/json');
    return response is DocumentResponse ? response : null;
  }

  /// Search knowledge
  Future<KnowledgeSearchResponse?> searchKnowledge(KnowledgeSearchRequest body) async {
    final response = await _client.post(ApiPaths.aiPath('/knowledge-bases/search'), body: body, contentType: 'application/json');
    return response is KnowledgeSearchResponse ? response : null;
  }

  /// List knowledge bases
  Future<KnowledgeBaseListResponse?> listKnowledgeBases(Map<String, dynamic>? params) async {
    final response = await _client.get(ApiPaths.aiPath('/knowledge-bases'), params: params);
    return response is KnowledgeBaseListResponse ? response : null;
  }

  /// Create knowledge base
  Future<KnowledgeBaseResponse?> createKnowledgeBase(KnowledgeBaseCreateRequest body) async {
    final response = await _client.post(ApiPaths.aiPath('/knowledge-bases'), body: body, contentType: 'application/json');
    return response is KnowledgeBaseResponse ? response : null;
  }

  /// Get document
  Future<DocumentResponse?> getDocument(String knowledgeBaseId, String documentId) async {
    final response = await _client.get(ApiPaths.aiPath('/knowledge-bases/${knowledgeBaseId}/documents/${documentId}'));
    return response is DocumentResponse ? response : null;
  }

  /// Delete document
  Future<void> deleteDocument(String knowledgeBaseId, String documentId) async {
    await _client.delete(ApiPaths.aiPath('/knowledge-bases/${knowledgeBaseId}/documents/${documentId}'));
  }

  /// Get knowledge base
  Future<KnowledgeBaseResponse?> getKnowledgeBase(String knowledgeBaseId) async {
    final response = await _client.get(ApiPaths.aiPath('/knowledge-bases/${knowledgeBaseId}'));
    return response is KnowledgeBaseResponse ? response : null;
  }

  /// Delete knowledge base
  Future<void> deleteKnowledgeBase(String knowledgeBaseId) async {
    await _client.delete(ApiPaths.aiPath('/knowledge-bases/${knowledgeBaseId}'));
  }
}

File: sdkwork-ai-sdk-python/sdkwork_ai_sdk/api/knowledge_bases.py
from typing import Any, Dict, List, Optional
from ..http_client import HttpClient
from ..models import DocumentAddRequest, DocumentResponse, KnowledgeBaseCreateRequest, KnowledgeBaseListResponse, KnowledgeBaseResponse, KnowledgeSearchRequest, KnowledgeSearchResponse, ListDocumentsResponse

class KnowledgeBasesApi:
    """Knowledge Bases API client."""
    
    def __init__(self, client: HttpClient):
        self._client = client

    def list_documents(self, knowledgeBaseId: str, params: Optional[Dict[str, Any]] = None) -> ListDocumentsResponse:
        """List documents"""
        return self._client.get(f"/ai/v3/v1/knowledge-bases/{knowledgeBaseId}/documents", params=params)

    def add_document(self, knowledgeBaseId: str, body: DocumentAddRequest) -> DocumentResponse:
        """Add document"""
        return self._client.post(f"/ai/v3/v1/knowledge-bases/{knowledgeBaseId}/documents", json=body)

    def search_knowledge(self, body: KnowledgeSearchRequest) -> KnowledgeSearchResponse:
        """Search knowledge"""
        return self._client.post(f"/ai/v3/v1/knowledge-bases/search", json=body)

    def list_knowledge_bases(self, params: Optional[Dict[str, Any]] = None) -> KnowledgeBaseListResponse:
        """List knowledge bases"""
        return self._client.get(f"/ai/v3/v1/knowledge-bases", params=params)

    def create_knowledge_base(self, body: KnowledgeBaseCreateRequest) -> KnowledgeBaseResponse:
        """Create knowledge base"""
        return self._client.post(f"/ai/v3/v1/knowledge-bases", json=body)

    def get_document(self, knowledgeBaseId: str, documentId: str) -> DocumentResponse:
        """Get document"""
        return self._client.get(f"/ai/v3/v1/knowledge-bases/{knowledgeBaseId}/documents/{documentId}")

    def delete_document(self, knowledgeBaseId: str, documentId: str) -> None:
        """Delete document"""
        return self._client.delete(f"/ai/v3/v1/knowledge-bases/{knowledgeBaseId}/documents/{documentId}")

    def get_knowledge_base(self, knowledgeBaseId: str) -> KnowledgeBaseResponse:
        """Get knowledge base"""
        return self._client.get(f"/ai/v3/v1/knowledge-bases/{knowledgeBaseId}")

    def delete_knowledge_base(self, knowledgeBaseId: str) -> None:
        """Delete knowledge base"""
        return self._client.delete(f"/ai/v3/v1/knowledge-bases/{knowledgeBaseId}")
